types/connect/connect-tests.ts

```diff
@@ -1,45 +1,65 @@
+// npm i --no-save connect
+// tsc --noEmit false && node connect-tests.js
+
 import * as http from "http";
 import connect = require("connect");
 
 const app = connect();
 
 // log all requests
-app.use((req: http.IncomingMessage, res: http.ServerResponse, next: connect.NextFunction) => {
-    console.log(req, res);
+function logReq(req: connect.Request, res: http.ServerResponse, next: connect.NextFunction) {
+    console.log(`${req.method} ${req.originalUrl}`);
     next();
-});
-
-// "Throw" an Error
-app.use((req: http.IncomingMessage, res: http.ServerResponse, next: connect.NextFunction) => {
-    next(new Error("Something went wrong!"));
-});
-
-// "Throw" a number
-app.use((req: http.IncomingMessage, res: http.ServerResponse, next: connect.NextFunction) => {
-    next(404);
-});
+}
 
 // Stop on errors
-app.use((err: any, req: http.IncomingMessage, res: http.ServerResponse, next: connect.NextFunction) => {
+function stopOnErr(err: any, req: connect.Request, res: http.ServerResponse, next: connect.NextFunction) {
     if (err) {
-        return res.end(`Error: ${err}`);
+        res.statusCode = +err || 500;
```

Comment:
This `+err` thing feels weird. If we decide to keep these changes, could we only set the status code if `typeof err === 'number'`?